tests/app/questionnaire/routing/test_operators.py

```diff
@@ -0,0 +1,291 @@
+from datetime import datetime, timedelta
+
+import pytest
+from dateutil.relativedelta import relativedelta
+
+from app.questionnaire.routing.helpers import datetime_as_midnight
+from app.questionnaire.routing.operator import Operator
+from app.questionnaire.rules import convert_to_datetime
+
+now = datetime.utcnow()
+now_as_yyyy_mm_dd = now.strftime("%Y-%m-%d")
+now_as_yyyy_mm = now.strftime("%Y-%m")
+now_as_yyyy = now.strftime("%Y")
+
+equals_operations = [
+    [(0.5, 0.5), True],
+    [(1.0, 1), True],
+    [(3, 3), True],
+    [(now, now), True],
+    [(0.5, 0.7), False],
+    [(1.0, 3), False],
+    [(3, 7), False],
+    [(now, datetime.utcnow()), False],
+    [("Yes", "Yes"), True],
+    [("CaseInsensitive", "caseInsensitive"), True],
+    [(None, None), True],
+    [(True, True), True],
+    [("Yes", "No"), False],
+    [(None, 1), False],
+    [(True, False), False],
+]
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    equals_operations,
+)
+def test_operation_equal(operands, expected_result):
+    operator = Operator(Operator.EQUAL)
+    assert operator.evaluate(operands) is expected_result
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    equals_operations,
+)
+def test_operation_not_equal(operands, expected_result):
+    operator = Operator(Operator.NOT_EQUAL)
+    assert operator.evaluate(operands) is not expected_result
+
+
+greater_than_and_less_than_operations = [
+    [(0.7, 0.5), True],
+    [(2, 1.0), True],
+    [(7, 3), True],
+    [(datetime.utcnow(), now), True],
```

Comment:
This is interesting as now = datetime.utcnow() I think we need to be a careful with this one, and do a proper offset

Reply:
So `now` is the time the test suite started, so a newly generated `datetime.utcnow()` would always be greater unless frozen. Have now assigned a future date using a delta to a variable for clarity.

Reply:
I realised that it would be higher, just prefer it to be explicit as to a less seasoned eye it does look a bit odd